src/app/loadout/LoadoutDrawer.tsx

```diff
@@ -381,17 +407,56 @@ class LoadoutDrawer extends React.Component<Props, State> {
       return;
     }
 
-    dimLoadoutService.saveLoadout(loadout).catch((e) => {
-      toaster.pop(
-        'error',
-        t('Loadouts.SaveErrorTitle'),
-        t('Loadouts.SaveErrorDescription', {
-          loadoutName: loadout.name,
-          error: e.message
-        })
-      );
-      console.error(e);
-    });
+    dimLoadoutService
+      .saveLoadout(loadout)
+      .then(this.handleLoadOutSaveResult)
+      .catch((e) => this.handleLoadoutError(e, loadout.name));
+  };
+
+  private handleLoadOutSaveResult = (clashingLoadout: Loadout | undefined) => {
+    if (clashingLoadout) {
+      this.showEditPopup(clashingLoadout);
+    } else {
+      this.close();
+    }
+  };
+
+  private showEditPopup(loadoutToEdit: Loadout) {
+    const onNameChange = this.changeNameHandler.bind(this);
+    const onEdit = () =>
+      this.setState({ loadout: loadoutToEdit, showEditPopup: false, isNew: false });
+
+    this.editPopup = (
```

Comment:
I'm not sure this needs to be an instance property - can't we just render this directly if `showEditPopup` is true, rather than saving it off to an instance property and rendering through that?